Dark mode friendly, broken down dashboards docs

bobbravo2 opened the pull request:
## Description
* Converted draw.io into mermaid (prop @guimou <3) 
* Broke the diagram into sub-headings
* Removed `.png` from the repo, open feedback re: if we want to keep the original draw.io, or if that's safe to remove, too. 


## How Has This Been Tested?
VSCode in light and dark modes. 

## Screenshots of the changes
![Screenshot 2025-03-25 at 15 41 48](https://github.com/user-attachments/assets/7ed2e8b1-2470-49d1-8ea0-ec18d3904648)
![Screenshot 2025-03-25 at 15 41 55](https://github.com/user-attachments/assets/9ce91b04-5287-46b4-a8ac-2fd751abd68c)





## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

christianvogt commented:
@bobbravo2 screenshots?

andrewballantyne reviewed (changes requested):
I'm going to see if I can play with this `mermaid` syntax to articulate the intended goal I was trying to articulate originally...

bobbravo2 commented on documentation/components/dashboard/README.md:
```diff
@@ -20,14 +20,63 @@ The Dashboard is the primary UI provided for OpenShift AI. The Dashboard serves
+    participant OAuthProxy as OAuth Proxy (Container)
+
+    EndUser->>DashboardUI: Request Dashboard HTML
+    DashboardUI->>OAuthProxy: (Not logged in) Redirect to OAuth Proxy
```
🤔 this is a great thought experiment! Thank you!

Would it help to clarify if we broke this specific diagram down two separate flows: 
1. when the `OAuth Proxy` call fails
2.  The user path for after a user has (successfully)  logged in?

andrewballantyne commented on documentation/components/dashboard/README.md:
````diff
@@ -20,14 +20,63 @@ The Dashboard is the primary UI provided for OpenShift AI. The Dashboard serves
+This section illustrates how an end user interacts with the Dashboard UI and the OAuth Proxy for authentication.
+
+```mermaid
+sequenceDiagram
+    participant EndUser as End User
+    participant DashboardUI as Dashboard UI (Browser)
+    participant OAuthProxy as OAuth Proxy (Container)
+
+    EndUser->>DashboardUI: Request Dashboard HTML
+    DashboardUI->>OAuthProxy: (Not logged in) Redirect to OAuth Proxy
+    EndUser->>OAuthProxy: Login via OAuth Proxy's Page
+    OAuthProxy->>DashboardUI: (Logged in) Redirect to Dashboard
````
Never used Mermaid before -- tried to futz with it and got something like this:

<img width="907" alt="image" src="https://github.com/user-attachments/assets/74f06bb6-6737-438f-92d8-65ed1ccb96ce" />


```suggestion
sequenceDiagram
    actor EndUser as End User
    participant Browser as Browser Window
    box Dashboard Pod
    participant OAuthProxy
    participant Dashboard
    end

    EndUser->>Browser: https://dashboard-route/*
    Browser--xOAuthProxy: (Not logged in)
    OAuthProxy-->>Browser: Return log in screen
    Browser-->>OAuthProxy: (log in)
    Browser<<->>Dashboard: (Logged in) Redirect to Dashboard
```

Feel free to modify this effort... I got the new content I used from this: https://mermaid.js.org/syntax/sequenceDiagram.html#messages

andrewballantyne reviewed (changes requested):
@bobbravo2 here is an example I think makes more sense... we may need to look at colours for the box, but it's not the end of the world without colour (text is in the middle of the two boxes so it sorta works even if you can't see the slightly off black line/background)

bobbravo2 commented:
@andrewballantyne thanks for the ideas, that's a lot simpler to parse! I've incorporated them and updated the main PR with the updated screenshots (in dark mode 🕶️ ) 

@christianvogt Screenshots added, too.
